tests/test_butler.py

```diff
@@ -649,6 +656,12 @@ def tearDown(self):
         bucket = s3.Bucket(self.bucketName)
         bucket.delete()
 
+        # unset any potentially set dummy credentials
+        keys = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
+        for key in keys:
+            if key in os.environ and "dummy" in os.environ[key]:
```

Comment:
I'm not a fan of guess work. I'd prefer the first setUp call to have set a property to indicate that dummy credentials are being used and then here do something like:
```python
if self.usedDummyCredentials:
    for key in ("AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"):
        del os.environ[key]
```

Reply:
`self.usedDummyCredentials` added.